### PR TITLE
Improve repo instruction tests

### PR DESCRIPTION
**Pull request by cosmok**
## Summary
- add edge case coverage for `getRepoInstructions`
- update instruction combination test to check order
- allow repo and folder lookup loop to consider root dir again

## Testing
- `npm test --silent`

------
https://chatgpt.com/codex/tasks/task_b_685fb99e6b94832ca0dea8df06fed1d4

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -279,10 +279,10 @@ async function getRepoInstructions(octokit, owner, repo, filePath, ref) {
   // Then look for the closest folder-level instructions (excluding repo root)
   let folderLevel = '';
   const parts = path.posix.dirname(filePath).split('/');
-  for (let i = parts.length; i > 0; i--) {
+  for (let i = parts.length; i >= 0; i--) {
```
[the remark this answers is not in the document]
Comment (cosmok) [reply index.js:282]: /review ELI5
